src/Network/NWIPMetadata.cs

```diff
@@ -0,0 +1,42 @@
+//
+// NWIPMetadata.cs: Bindings the Netowrk nw_protocol_metadata_t API that is an IP.
+//
+// Authors:
+//   Manuel de la Pena <mandel@microsoft.com>
+//
+// Copyrigh 2019 Microsoft
+//
+using System;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+namespace Network {
+
+	[TV (12,0), Mac (10,14), iOS (12,0), Watch (6,0)]
+	public class NWIPMetadata : NWProtocolMetadata {
+
+		internal NWIPMetadata (IntPtr handle, bool owns) : base (handle, owns) {}
+
+		public NWIPMetadata () : this (nw_ip_create_metadata (), owns: true) {}
+
+		public NWIPEcnFlag EcnFlag {
+			get => nw_ip_metadata_get_ecn_flag (GetCheckedHandle ());
+			set => nw_ip_metadata_set_ecn_flag (GetCheckedHandle (), value);
+		}
+
+		// A single tick represents one hundred nanoseconds, API returns: the time at which a packet was received, in nanoseconds
+		// so we get nanoseconds, divicde by 100 and use from ticks
+		public TimeSpan ReceiveTime {
+			get {
+				var time = nw_ip_metadata_get_receive_time (GetCheckedHandle ());
+				return (time == 0) ? TimeSpan.Zero : TimeSpan.FromTicks ((long) time / 100);
```

Comment:
I don't think this is needed. There's no indication that `FromTicks (0)` returns an illegal value. I looked at the MS reference source for `FromTicks` and it returns `new TimeSpan (ticks)`. Since `TimeSpan` is a struct, this is cheap.

Reply:
More of not using a new struct every time, and use the static one. Can be removed.

Reply:
The internal representation is a long. Returning a long-sized struct is essentially free.

Reply:
Removed it, so not a big issue.